cmd/zfs/zfs_main.c

```diff
@@ -8350,3 +8381,69 @@ main(int argc, char **argv)
 
 	return (ret);
 }
+
+#ifdef __FreeBSD__
+#include <sys/jail.h>
+#include <jail.h>
+/*
+ * Attach/detach the given dataset to/from the given jail
+ */
+/* ARGSUSED */
+static int
+do_jail(int argc, char **argv, int attach)
```

Comment:
Can you rename this function `zfs_do_jail_impl()` for consistency with `zfs_do_allow_unallow_impl`, `zfs_do_hold_rele_impl`, etc.  It would be nice to switch `int attach` to `boolean_t attach` as well.